modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/ResolverSet.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.resolver;
+
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.context.MuleContextAware;
+import org.mule.api.lifecycle.Lifecycle;
+import org.mule.extension.introspection.Parameter;
+import org.mule.module.extension.internal.runtime.ObjectBuilder;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * A {@link ValueResolver} which is based on associating a set of
+ * {@link Parameter}s -&gt; {@link ValueResolver} pairs. The result
+ * of evaluating this resolver is a {@link ResolverSetResult}.
+ * <p/>
+ * The general purpose of this class is to repeatedly evaluate a set of {@link ValueResolver}s
+ * which results are to be used in the construction of an object, so that the structure
+ * of such can be described only once (by the set of {@link Parameter}s and {@link ValueResolver}s
+ * but evaluated many times. With this goal in mind is that the return value of this resolver
+ * will always be a {@link ResolverSetResult} which then can be used by a {@link ObjectBuilder}
+ * to generate an actual object.
+ * <p/>
+ * Instances of this class are to be considered thread safe and reusable
+ *
+ * @since 3.7.0
+ */
+public class ResolverSet implements ValueResolver<Object>
+{
+
+    private Map<Parameter, ValueResolver> resolvers = new LinkedHashMap<>();
+    private boolean dynamic = false;
+
+    /**
+     * Links the given {@link ValueResolver} to the given {@link Parameter}.
+     * If such {@code parameter} was already added, then the associated {@code resolver}
```

Comment:
what value would that add? Since this is an internal class, why couldn't we have some component that does post processing and replaces the value?